worker/src/features/evaluation/evalService.ts

```diff
@@ -319,18 +342,77 @@ export const evaluate = async ({
   // persist the score and update the job status
   const scoreId = randomUUID();
 
+  const baseScore = {
+    id: scoreId,
+    traceId: job.job_input_trace_id,
+    name: config.score_name,
+    value: parsedLLMOutput.score,
+    comment: parsedLLMOutput.reasoning,
+    source: ScoreSource.EVAL,
+  };
+
   await prisma.score.create({
     data: {
-      id: scoreId,
-      traceId: job.job_input_trace_id,
-      name: config.score_name,
-      value: parsedLLMOutput.score,
-      comment: parsedLLMOutput.reasoning,
-      source: "EVAL",
+      ...baseScore,
       projectId: event.projectId,
     },
   });
 
+  // Write score to S3 and ingest into queue for Clickhouse processing
+  try {
+    if (env.LANGFUSE_S3_EVENT_UPLOAD_ENABLED === "true" && env.CLICKHOUSE_URL) {
+      if (env.LANGFUSE_S3_EVENT_UPLOAD_BUCKET === undefined) {
+        throw new Error("S3 event store is enabled but no bucket is set");
+      }
+      const s3Client = getS3StorageServiceClient(
+        env.LANGFUSE_S3_EVENT_UPLOAD_BUCKET,
+      );
+      await s3Client.uploadJson(
+        `${env.LANGFUSE_S3_EVENT_UPLOAD_PREFIX}${event.projectId}/score/${scoreId}/${randomUUID()}.json`,
+        [
+          {
+            id: randomUUID(),
+            timestamp: new Date().toISOString(),
+            type: eventTypes.SCORE_CREATE,
+            body: {
+              ...baseScore,
+              dataType: "NUMERIC",
+            },
+          },
+        ],
+      );
+
```

Comment:
Curiosity question: why do we chose to not queue the score create event if we fail to upload the score json to s3? Is this a general requirement we are enforcing across the platform now?

Reply:
Awesome, thanks for the detailed context! 🙌 Makes sense